test/e2e/unmanaged_dns_test.go

```diff
@@ -0,0 +1,380 @@
+		t.Fatalf("failed to get LoadBalancer service: %v", err)
+	}
+
+	if !ingresscontroller.IsServiceInternal(lbService) {
+		t.Fatalf("load balancer %s is external but should be internal", lbService.Name)
+	}
+
+	wildcardRecordName := controller.WildcardDNSRecordName(ic)
+	wildcardRecord := &iov1.DNSRecord{}
+	if err := kclient.Get(context.TODO(), wildcardRecordName, wildcardRecord); err != nil {
+		t.Fatalf("failed to get wildcard dnsrecord %s: %v", wildcardRecordName, err)
+	}
+
+	if wildcardRecord.Spec.DNSManagementPolicy != iov1.UnmanagedDNS {
+		t.Fatalf("DNSRecord %s expected in UnmanagedDNS state but got %s", wildcardRecordName.Name, wildcardRecord.Spec.DNSManagementPolicy)
+	}
+
+	routerDeployment := &appsv1.Deployment{}
+	if err := kclient.Get(context.TODO(), controller.RouterDeploymentName(ic), routerDeployment); err != nil {
+		t.Fatalf("failed to get router deployment: %v", err)
+	}
+
+	verifyUnmanagedInternalIngressController(t, name, "apps."+ic.Spec.Domain, wildcardRecord.Spec.Targets[0], routerDeployment.Spec.Template.Spec.Containers[0].Image)
+
+	t.Logf("Updating ingresscontroller %s to Managed", ic.Name)
+
+	if err := updateIngressControllerSpecWithRetryOnConflict(t, name, 5*time.Minute, func(ics *operatorv1.IngressControllerSpec) {
+		ics.EndpointPublishingStrategy.LoadBalancer.Scope = operatorv1.ExternalLoadBalancer
+		ics.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy = operatorv1.ManagedLoadBalancerDNS
+	}); err != nil {
+		t.Fatalf("failed to update ingresscontroller %s: %v", name, err)
+	}
+
+	if err := kclient.Delete(context.TODO(), lbService); err != nil && !errors.IsNotFound(err) {
+		t.Fatalf("failed to delete svc %s: %v", lbService.Name, err)
+	}
+
+	t.Log("Waiting for stable conditions on ingresscontroller after Managed")
+
+	// Wait for the load balancer and DNS to reach stable conditions.
+	if err := waitForIngressControllerCondition(t, kclient, 10*time.Minute, name, availableConditionsForIngressControllerWithLoadBalancer...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	if !ingresscontroller.IsServiceInternal(lbService) {
+		t.Fatalf("load balancer %s is internal but should be external", lbService.Name)
+	}
+
+	if err := kclient.Get(context.TODO(), wildcardRecordName, wildcardRecord); err != nil {
+		t.Fatalf("failed to get wildcard dnsrecord %s: %v", wildcardRecordName, err)
+	}
+
+	if wildcardRecord.Spec.DNSManagementPolicy != iov1.ManagedDNS {
+		t.Fatalf("DNSRecord %s expected in ManagedDNS state but got %s", wildcardRecordName.Name, wildcardRecord.Spec.DNSManagementPolicy)
+	}
+
+	if len(wildcardRecord.Status.Zones) != 2 {
+		t.Fatalf("DNSRecord %s expected allocated dnsZones but found none", wildcardRecordName.Name)
+	}
+
+	verifyUnmanagedIngressController(t, name, "apps."+ic.Spec.Domain, wildcardRecord.Spec.Targets[0])
+}
+
+func verifyUnmanagedIngressController(t *testing.T, name types.NamespacedName, hostname, address string) {
```

Comment:
Is this really testing UnmanagedDNS? Would it make sense to add `DNS` in the function name to just be a little clearer?